samples/bluetooth/peripheral_hr/src/main.c

```diff
@@ -44,11 +44,32 @@ static void disconnected(struct bt_conn *conn, uint8_t reason)
 	printk("Disconnected (reason 0x%02x)\n", reason);
 }
 
+static void security_changed(struct bt_conn *conn, bt_security_t level, enum bt_security_err err)
+{
+	printk("Security changed: conn %p, level %d err %d\n", conn, level, err);
+}
+
 BT_CONN_CB_DEFINE(conn_callbacks) = {
 	.connected = connected,
 	.disconnected = disconnected,
+	.security_changed = security_changed,
 };
 
+static void att_chan_connected(struct bt_conn *conn, uint16_t cid, uint16_t mtu, uint16_t mps)
+{
+	printk("ATT channel connected. conn: %p, cid: 0x%04X, mtu: %d. mps: %d\n", conn, cid, mtu,
+	       mps);
+}
+
+static void att_chan_disconnected(struct bt_conn *conn, uint16_t cid)
+{
+	if (cid) {
+		printk("ATT channel with cid 0x%04X disconnected\n", cid);
+	} else {
+		printk("Connection of ATT channel failed\n");
```

Comment:
Disconnection

Reply:
The L2CAP channel disconnected callback is called when it is deleted (`bt_l2cap_chan_del`). This also happens when ecred connection fails. Thus, this callback is called with `cid=0x0000` if connection failed. This may not be the most intuitive and we may want to change this.